libfwupd/fwupd-remote.c

```diff
@@ -185,7 +187,18 @@ fwupd_remote_set_id (FwupdRemote *self, const gchar *id)
 	g_strdelimit (priv->id, ".", '\0');
 }
 
-static void
+/**
+ * fwupd_remote_set_filename_source:
+ * @self: a #FwupdRemote
+ * @filename_source: (nullable): The mutable filename
+ *
+ * Sets the filename of the mutable metadata source.
+ * This is typically a location in /var/lib/fwupd/remotes.d rather than the
+ * possibly-read-only location in /etc/fwupd/remotes.d
+ *
+ * Since: 1.6.1
+ **/
+void
```

Comment:
so going this approach a single field can't be modified, but rather you need to fully duplicate the file, right?